docker_daemon/check.py

```diff
@@ -637,14 +638,46 @@ def _report_cgroup_metrics(self, container, tags):
                         if value is not None:
                             metric_func(self, mname, value, tags=tags)
 
+    def _build_network_mapping(self, container):
+        """Matches /proc/$PID/net/route and docker inspect to map interface names to docker network name"""
+        try:
+            proc_net_route_file = os.path.join(container['_proc_root'], 'net/route')
+
+            docker_gateways = {}
+            for netname, netconf in container.get(u'NetworkSettings').get(u'Networks').iteritems():
+                docker_gateways[netname] = struct.unpack('<L', socket.inet_aton(netconf.get(u'Gateway')))[0]
+
+            mapping = {}
+            with open(proc_net_route_file, 'r') as fp:
+                lines = fp.readlines()
+                for l in lines[1:]:
+                    cols = l.split()
+                    if cols[1] == '00000000':
+                        continue
+                    destination = int(cols[1], 16)
+                    for net, gw in docker_gateways.iteritems():
+                        if gw & destination == destination:
```

Comment:
I don't think this wouldn't be strictly true for regular networks in the general case, ie. you could have a destination 172.18.0.0/16 that is reachable via a router/gw in 10.0.0.1, and have eth0 on 10.0.0.0/24 and `172.18.0.0/16` would still be part of eth0 networks.